src/scanner.py

```diff
@@ -127,10 +127,20 @@ def scan_token(self):
             self.identifier()
 
         else:
+            
             # TODO: pythonic error
-            # self.error(self.line, 'unexpected character')
-            print("error")
-
+            # class Error(Exception):
+
+            # raise Error
+            #FIX THIS ERROR
+            class Error(Exception):
```

Comment:
You can move the `Error` class outside the `Scanner` class. It would be better to rename the error, the error name which relates to scanner would be good. Do we need `def __init__(...)`? I think inheriting the Exception takes care of that.

Reply:
Thanks for your suggestions @twentyse7en  :D.